### PR TITLE
[ISSUE #4938, ISSUE #4939] fix the concurrency problems and limit the times of renew when some error occurred in ReceiptHandleProcessor

### PR DESCRIPTION
**Pull request by xdkxlk**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

1. fix the concurrency problems when renew and ack/changeInvisibleTime at the same time
2. retry renew for a limited times when some error occurred
3. code refactor for ReceiptHandleProcessor

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by aaron-ai** [comment]
image

@xdkxlk thx for contributing, looks like another flaky test, could you create an issue for this?

File: proxy/src/main/java/org/apache/rocketmq/proxy/common/MessageReceiptHandle.java [hunk 1]
```diff
@@ -27,24 +29,26 @@ public class MessageReceiptHandle {
...
-        long queueOffset, int reconsumeTimes, long expectInvisibleTime) {
+    public MessageReceiptHandle(String group, String topic, int queueId, String receiptHandleStr, String messageId,
+        long queueOffset, int reconsumeTimes) {
+        ReceiptHandle receiptHandle = ReceiptHandle.decode(receiptHandleStr);
         this.group = group;
         this.topic = topic;
         this.queueId = queueId;
-        this.receiptHandle = receiptHandle;
-        this.originalReceiptHandle = receiptHandle;
+        this.receiptHandle = receiptHandleStr;
```
Comment (lollipopjin) [inline proxy/src/main/java/org/apache/rocketmq/proxy/common/MessageReceiptHandle.java:45]: Can here rename to receiptHandleStr for avoiding confusion with ReceiptHandle receiptHandle

Comment (xdkxlk) [reply proxy/src/main/java/org/apache/rocketmq/proxy/common/MessageReceiptHandle.java]: done

File: proxy/src/test/java/org/apache/rocketmq/proxy/common/ReceiptHandleGroupTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                    }
+                    messageReceiptHandle.updateReceiptHandle(handle2);
+                    return FutureUtils.addExecutor(CompletableFuture.completedFuture(messageReceiptHandle), Executors.newCachedThreadPool());
+                });
+            } catch (Exception ignored) {
+            }
+        }, "computeThread");
+        removeThread.start();
+        computeThread.start();
+        Thread.sleep(10);
```
Comment (aaron-ai) [inline proxy/src/test/java/org/apache/rocketmq/proxy/common/ReceiptHandleGroupTest.java:98]: Could we refine the sleep here?

Comment (xdkxlk) [reply proxy/src/test/java/org/apache/rocketmq/proxy/common/ReceiptHandleGroupTest.java]: done

**Review by lollipopjin — APPROVED** [review]
LGTM

**Pull request merged**